source/common/upstream/health_checker_impl.cc

```diff
@@ -374,7 +411,10 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onResponseComplete() {
     handleSuccess(true);
     break;
   case HealthCheckResult::Failed:
-    handleFailure(envoy::data::core::v3::ACTIVE);
+    handleFailure(envoy::data::core::v3::ACTIVE, false);
+    break;
+  case HealthCheckResult::Retriable:
+    handleFailure(envoy::data::core::v3::ACTIVE, true);
```

Comment:
Might want to comment the boolean parameter for extra clarity. Something like  `, /*retriable=*/true);`